src/MiniJuvix/Translation/MonoJuvixToMiniC.hs

```diff
@@ -144,25 +217,72 @@ mkName n =
     idSuffix :: Text
     idSuffix = "_" <> show (n ^. Mono.nameId . unNameId)
 
-goFunctionDef :: Mono.FunctionDef -> [CCode]
-goFunctionDef Mono.FunctionDef {..} =
-  [ ExternalFunc
-      ( Function
-          { _funcReturnType = funReturnType ^. typeDeclType,
-            _funcIsPtr = funReturnType ^. typeIsPtr,
-            _funcQualifier = None,
-            _funcName = funcName,
-            _funcArgs = namedArgs asFunArg funArgTypes,
-            _funcBody = maybeToList (BodyStatement <$> mkBody (goFunctionClause <$> toList _funDefClauses))
-          }
-      )
-  ]
+cFunTypeToFunSig :: Text -> CFunType -> FunctionSig
+cFunTypeToFunSig name CFunType {..} =
+  FunctionSig
+    { _funcReturnType = _cFunReturnType ^. typeDeclType,
+      _funcIsPtr = _cFunReturnType ^. typeIsPtr,
+      _funcQualifier = None,
+      _funcName = name,
+      _funcArgs = namedArgs asFunArg _cFunArgTypes
+    }
+
+genFunctionSig :: Mono.FunctionDef -> [CCode]
+genFunctionSig Mono.FunctionDef {..} =
+  [ExternalFuncSig (cFunTypeToFunSig funName funType)]
     <> (ExternalMacro . CppDefineParens <$> toList nullaryDefine)
+  where
+    funType :: CFunType
+    funType = typeToFunType _funDefType
+    isNullary :: Bool
+    isNullary = null (funType ^. cFunArgTypes) && funcBasename /= Str.main_
```

Comment:
This function may deserve a better place. A similar definition, also named, `isNullary` is used below.
Perhaps you want to define `isNullary :: CFunType -> Bool`.